src/main/java/org/sunbird/bpreports/service/BPReportConsumer.java

```diff
@@ -0,0 +1,457 @@
+package org.sunbird.bpreports.service;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.poi.ss.usermodel.*;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.MatchQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.search.sort.SortOrder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.ObjectUtils;
+import org.sunbird.bpreports.postgres.entity.WfStatusEntity;
+import org.sunbird.bpreports.postgres.repository.WfStatusEntityRepository;
+import org.sunbird.cassandra.utils.CassandraOperation;
+import org.sunbird.common.model.SBApiResponse;
+import org.sunbird.common.service.OutboundRequestHandlerServiceImpl;
+import org.sunbird.common.util.CbExtServerProperties;
+import org.sunbird.common.util.Constants;
+import org.sunbird.common.util.IndexerService;
+import org.sunbird.storage.service.StorageService;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.*;
+
+@Component
+public class BPReportConsumer {
+
+    private static final Logger logger = LoggerFactory.getLogger(BPReportConsumer.class);
+
+    @Autowired
+    ObjectMapper mapper;
+
+
+    @Autowired
+    WfStatusEntityRepository wfStatusEntityRepository;
+
+    @Autowired
+    CassandraOperation cassandraOperation;
+
+    @Autowired
+    IndexerService indexerService;
+
+    @Autowired
+    CbExtServerProperties serverProperties;
+
+    @Autowired
+    StorageService storageService;
+
+
+    @KafkaListener(topics = "${kafka.topic.bp.report}", groupId = "${kafka.topic.bp.report.group}")
+    private void initiateBPReportGeneration(ConsumerRecord<String, String> data) {
+        logger.info("AssessmentAsyncSubmitConsumer::processMessage.. started.");
+        try {
+            Map<String, Object> asyncRequest = mapper.readValue(data.value(), new TypeReference<Map<String, Object>>() {
+            });
+            generateBPReport(asyncRequest);
```

Comment:
this should be thread ??
how long it would take to generate this request... if there is a delay -- kafka messages won't be acknowledged - what will happen in that case?